crypto/evp/evp_lib.c

```diff
@@ -1058,6 +1063,33 @@ EVP_MD *EVP_MD_CTX_get1_md(EVP_MD_CTX *ctx)
     return md;
 }
 
+int EVP_MD_CTX_get_size_ex(const EVP_MD_CTX *ctx)
+{
+    const EVP_MD *md = EVP_MD_CTX_get0_md(ctx);
+
+    if (EVP_MD_xof(md)) {
+        if (EVP_MD_is_a(md, "SHAKE128")
+                || EVP_MD_is_a(md, "SHAKE256")) {
```

Comment:
Instead of is_a() check for concrete hash, I would attempt calling the EVP_MD_CTX_get_params() and only if that fails I would fall back to EVP_MD_get_size(md).

This will be needed for FIPS provider compatibility anyway as SHAKE in old providers won't support EVP_MD_CTX_get_params() anyway.

Other option would be to not do the fallback and just fail. Then the old provider compatibility will be broken. But it can be argued that calling EVP_MD_get_size() here for XOF functions is actually incorrect in all cases.

Reply:
Gost tests are checking this value..

Reply:
Unless EVP_MD_get_size() is interpreted to mean the "default" value

Reply:
> Unless EVP_MD_get_size() is interpreted to mean the "default" value

I'd suggest this.